apisix/init.lua

```diff
@@ -387,6 +387,13 @@ function _M.http_access_phase()
     end
 
     local up_id = route.value.upstream_id
+
+    -- used for the traffic-split plugin
+    if route.value.upstream_id_tmp then
```

Comment:
Maybe we can set & use `ctx.upstream_id` directly. No more modification to the route.

Reply:
Do you mean to directly set the variable of `ctx.upstream_id` in the traffic-split plugin to cache the upstream_id in the plugin?

Reply:
Instead of setting `ctx.matched_route.value.upstream_id_tmp`, can we set `ctx.upstream_id` directly?

Reply:
This can directly set `ctx.upstream_id`, I will update the code.

Reply:
updated.